src/formState.test.tsx

```diff
@@ -1615,16 +1632,19 @@ describe("formState", () => {
     // When the data/child is now available
     click(r.refreshData);
     // And the field is blurred
-    click(r.blur);
-    // Then expect onBlur to triggered
-    expect(onBlur).toBeCalledTimes(1);
```

Comment:
Per the updated comment, this going from 1 to 0 I think is actually a bug fix, i.e. we were triggering an `onBlur` before when the list didn't actually change.

The `hasNewItems` code change that was needed for the new test case ended up fixing this as well.